stan/math/prim/scal/meta/contains_fvar.hpp

```diff
@@ -3,25 +3,16 @@
 
 #include <stan/math/prim/scal/meta/is_fvar.hpp>
 #include <stan/math/prim/scal/meta/scalar_type.hpp>
+#include <stan/math/prim/scal/meta/or.hpp>
 
 namespace stan {
 
 /**
- * Metaprogram to calculate the base scalar return type resulting
- * from promoting all the scalar types of the template parameters.
+ * Metaprogram to determine if any of the
+ * provided types is a fvar.
  */
-template <typename T1, typename T2 = double, typename T3 = double,
-          typename T4 = double, typename T5 = double, typename T6 = double>
-struct contains_fvar {
-  enum {
-    value = is_fvar<typename scalar_type<T1>::type>::value
-            || is_fvar<typename scalar_type<T2>::type>::value
-            || is_fvar<typename scalar_type<T3>::type>::value
-            || is_fvar<typename scalar_type<T4>::type>::value
-            || is_fvar<typename scalar_type<T5>::type>::value
-            || is_fvar<typename scalar_type<T6>::type>::value
-  };
-};
+template <typename... T>
+using contains_fvar = math::or_<is_fvar<typename scalar_type<T>::type>...>;
 
 }  // namespace stan
 #endif
```

Comment:
[question]
Was there no need to change contains_fvar_test?

Reply:
Not really, the API remained the same. I added a few tests for more than 1 type.